lib/g_searcher_web/controllers/api/report_controller.ex

```diff
@@ -0,0 +1,30 @@
+defmodule GSearcherWeb.API.ReportController do
+  use GSearcherWeb, :controller
+
+  alias GSearcher.Reports
+  alias GSearcherWeb.ErrorHandler
+  alias GSearcherWeb.Validators.API.CreateReportParams
+  alias GSearcherWeb.Validators.ParamsValidator
+
+  def create(conn, params) do
+    %{id: user_id} = conn.assigns.user
+
+    with {:ok, %{csv: csv}} <-
+           ParamsValidator.validate(params, as: CreateReportParams),
+         {:ok, _report} <- Reports.create_report(user_id, csv.filename, csv.path) do
+      send_resp(conn, :no_content, "")
+    else
+      {:error, :invalid_params, %Ecto.Changeset{} = report_changeset} ->
+        changeset_errors = ErrorHandler.build_changeset_error_message(report_changeset)
+
+        conn
+        |> put_status(:bad_request)
+        |> ErrorHandler.render_error_json(:bad_request, changeset_errors)
+
+      {:error, :failed_to_save_keywords} ->
+        conn
+        |> put_status(:internal_server_error)
```

Comment:
this is not a true `internal_server_error` it's because of the invalid data or somehow because the user's data there and we could catch that error as `{:error, :failed_to_save_keywords}` which mean the error was identified, so it's not an internal error, instead we can return it as 422 status code, https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422

Reply:
Yeah, that makes sense. 😄 
I've updated the HTTP statuses in cb176bb9abcba5bceb7d06633bb0a7473a9111ab 👍